src/middleware/isAdmin.middleware.ts

```diff
@@ -0,0 +1,22 @@
+import { NextFunction, Request, Response } from 'express';
+import { verify } from 'jsonwebtoken';
+import UserModel from '../models/user.model';
+
+export const isAdmin = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  const token = req.headers['authorization'];
+  if (!token) return res.status(401).send('No User');
+  try {
+    const verified = verify(token, process.env.JWT_SECRET!);
+    req.user = verified;
```

Comment:
Please use verifyJWT method from jwt.util file because this access token will be signed by a private key which uses RSA algorithm and my verifyJWT method will verify the access token with private key